proxy/src/main/java/com/wavefront/agent/QueuedAgentService.java

```diff
@@ -385,6 +312,163 @@ public Long value() {
+                    }
+                    break;
+                  } else {
+                    logger.log(Level.WARNING, "[RETRY THREAD " + threadId + "] cannot submit data to Wavefront servers. Will " +
+                        "re-attempt later", Throwables.getRootCause(ex));
+                  }
+                // this can potentially cause a duplicate task to be injected (but since submission is mostly
+                // idempotent it's not really a big deal)
+                task.service = null;
+                task.currentAgentId = null;
+                taskQueue.add(task);
+                queuePointsCount.addAndGet(taskSize);
+                if (failures > 10) {
+                  logger.warning("[RETRY THREAD " + threadId + "] saw too many submission errors. Will " +
+                      "re-attempt later");
+                  break;
+                }
+              } finally {
+                if (removeTask) {
+                  taskQueue.remove();
+                  queuePointsCount.addAndGet(-taskSize);
+                }
+              }
+            } finally {
+              taskQueue.getLockObject().unlock();
+            }
+          }
+        } catch (Throwable ex) {
+          logger.log(Level.WARNING, "[RETRY THREAD " + threadId + "] unexpected exception", ex);
+        } finally {
+          if (rateLimiting) {
+            logger.fine("[RETRY THREAD " + threadId + "] Successful Batches: " + successes +
+                ", Failed Batches: " + failures);
+            logger.fine("[RETRY THREAD " + threadId + "] Rate limit reached, will re-attempt later");
+            // if proxy rate limit exceeded, try again in 250..500ms (to introduce some degree of fairness)
+            executorService.schedule(this, 250 + (int) (Math.random() * 250), TimeUnit.MILLISECONDS);
+          } else {
+            if (successes == 0 && failures != 0) {
+              backoffExponent = Math.min(4, backoffExponent + 1); // caps at 2*base^4
+            } else {
+              backoffExponent = 1;
+            }
+            long next = (long) ((Math.random() + 1.0) *
+                Math.pow(retryBackoffBaseSeconds.get(), backoffExponent));
+            logger.fine("[RETRY THREAD " + threadId + "] Successful Batches: " + successes +
+                ", Failed Batches: " + failures);
+            logger.fine("[RETRY THREAD " + threadId + "] RESCHEDULING in " + next);
+            executorService.schedule(this, next, TimeUnit.SECONDS);
+          }
+        }
+      }
+    };
+  }
+
+  private ObjectQueue<ResubmissionTask> createTaskQueue(final UUID agentId, File buffer) throws
+      IOException {
+    return new FileObjectQueue<>(buffer,
+        new FileObjectQueue.Converter<ResubmissionTask>() {
+          @Override
+          public ResubmissionTask from(byte[] bytes) throws IOException {
+            try {
+              if (bytes.length > 2 && bytes[0] == (byte) 0x1f && bytes[1] == (byte) 0x8b) {
```

Comment:
you can probably remove logic that does this (for that matter, we should remove this logic completely) since this is for backwards compatibility and we shouldn't need any